### PR TITLE
feat: logout from multiple accounts

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-976/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/feature/AccountSwitchUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import com.wire.kalium.logic.feature.session.UpdateCurrentSessionUseCase
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Suppress("LongParameterList")
+@Singleton
+class AccountSwitchUseCase @Inject constructor(
+    private val updateCurrentSessionUseCase: UpdateCurrentSessionUseCase,
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/feature/AccountSwitchUseCase.kt]:
```suggestion
    private val updateCurrentSession: UpdateCurrentSessionUseCase,
```

File: app/src/main/kotlin/com/wire/android/feature/AccountSwitchUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    private val navigationManager: NavigationManager,
+    private val getSessionsUseCase: GetSessionsUseCase,
+    private val getCurrentSessionUseCase: CurrentSessionUseCase,
+    private val deleteSessionUseCase: DeleteSessionUseCase,
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/feature/AccountSwitchUseCase.kt]:
```suggestion
    private val getSessions: GetSessionsUseCase,
    private val getCurrentSession: CurrentSessionUseCase,
    private val deleteSession: DeleteSessionUseCase,
```

File: app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt [hunk 3]
```diff
@@ -64,8 +64,8 @@ class SelfUserProfileViewModel @Inject constructor(
     private val selfServerLinks: SelfServerConfigUseCase,
     private val kaliumConfigs: KaliumConfigs,
     private val otherAccountMapper: OtherAccountMapper,
-    private val updateCurrentSession: UpdateCurrentSessionUseCase,
-    private val observeEstablishedCalls: ObserveEstablishedCallsUseCase
+    private val observeEstablishedCalls: ObserveEstablishedCallsUseCase,
+    private val accountSwitchUseCase: AccountSwitchUseCase
```
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt]:
```suggestion
    private val accountSwitch: AccountSwitchUseCase
```

**Review by yamilmedina — APPROVED** [review]
Small changes for naming use cases, but other than that let's go !

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-976/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-976/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-976/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4419276)(4419276)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4419276)(4419276)-dev-release.apk)

**Pull request merged**